src/write.c

```diff
@@ -878,7 +879,36 @@ avifResult avifEncoderAddImageGrid(avifEncoder * encoder,
     if ((gridCols == 0) || (gridCols > 256) || (gridRows == 0) || (gridRows > 256)) {
         return AVIF_RESULT_INVALID_IMAGE_GRID;
     }
-    return avifEncoderAddImageInternal(encoder, gridCols, gridRows, cellImages, 1, addImageFlags | AVIF_ADD_IMAGE_FLAG_SINGLE); // only single image grids are supported
+    return avifEncoderAddImageInternal(encoder, gridCols, gridRows, 1, cellImages, 1, addImageFlags | AVIF_ADD_IMAGE_FLAG_SINGLE); // only single image grids are supported
+}
+
+
+avifResult avifEncoderAddImageProgressive(avifEncoder * encoder,
+                                          uint32_t layerCount,
+                                          const avifImage * const * layerImages,
+                                          avifAddImageFlags addImageFlags)
+{
+    avifDiagnosticsClearError(&encoder->diag);
+    if ((layerCount == 0) || (layerCount > MAX_AV1_LAYER_COUNT)) {
```

Comment:
This check and those in `avifEncoderAddImageProgressiveGrid()` could be moved to `avifEncoderAddImageInternal()`